MHTimer.js

```diff
@@ -2458,42 +2525,131 @@ function updRH(message) {
+        if (relic_hunter.location != new_location) {
+            relic_hunter.location = new_location
+            setImmediate(remindRH, new_location);
+        }
         console.log(`Now in ${relic_hunter.location}`);
-        relic_hunter.source = "webhook";
+        relic_hunter.source = 'webhook';
     }
 }
 
 /**
- * Processes a request to find the relic hunter
- * @param {TextChannel} channel the channel on which to respond.
+ * Especially at startup, find the relic hunter's location
+ * TODO: This might replace the reset function
  */
-function findRH(channel) {
-    let responseStr = "";
-    //RH was not seen since beginning of today, grab the info
-    let req = request({
+async function getRHLocation() {
+    console.log(`Relic hunter was in ${relic_hunter.location} according to ${relic_hunter.source}`);
+    const newLocations = await Promise.all([
+        DBGamesRHLookup(),
+        MHCTRHLookup()
+    ]);
+    // Precedence goes to MHCT because it would have seen RH
+    let newLocation = newLocations.find(t => t.source === 'MHCT' && t.location !== 'unknown');
+    if (newLocation && newLocation.location) {
+        relic_hunter.location = newLocation.location;
+        relic_hunter.source = 'MHCT';
+    } else {
+        newLocation = newLocations.find(t => t.source === 'DBGames');
+        if (newLocation && newLocation.location) {
+            relic_hunter.location = newLocation.location;
+            relic_hunter.source = 'DBGames';
+        }
+    }
+    console.log(`Relic hunter is in ${relic_hunter.location} according to ${relic_hunter.source}`);
+}
+
+/**
+ * Looks up Relic Hunter Location from DBGames via Google Sheets
+ * @returns {Promise<unknown>}
+ * @constructor
+ */
+function DBGamesRHLookup() {
+    let new_location = 'unknown';
+    return new Promise(function(resolve){
+        let req = request({
+                uri: 'https://docs.google.com/spreadsheets/d/e/2PACX-1vSsqAjocBWcN5dDLXuOBfnBhyrTaO7ZeIEAFlDnQ4r6zqcvtuLKMDBQCh5I8-3M9irS4-17OPfvgKtY/pub?gid=1975888453&single=true&output=csv'
+            }, (error, response, body) => {
+                if (!error && response.statusCode === 200 && body) {
+                    console.log(`Relic hunter location updated from ${relic_hunter.location} to ${body}`);
+                    resolve({
+                        source: 'DBGames',
+                        location: body,
+                        last_seen: 0
+                    });
+                } else {
+                    console.log(`RelicHunter query failed:`, error, response, body);
+                    resolve({
+                        source: 'broken',
+                        location: 'unknown',
+                        last_seen: 0
+                    });
+                }
+            });
+    });
+
+}
+
+/**
+ * Looks up the relic hunter location from MHCT
+ */
+async function MHCTRHLookup() {
+    if (relic_hunter.source === 'MHCT' && relic_hunter.location !== 'unknown') {
+        return;
```

Comment:
To exit quickly here you should return a resolved promise: `return Promise.resolve({ source: 'MHCT', ..... });`